koala/ast/graph.py

```diff
@@ -1078,45 +1106,62 @@ def gen_graph(self, outputs = None, inputs = None):
                 # if the dependency is a multi-cell range, create a range object
                 elif is_range(dep):
 
+                    # In opposition with previous version,
+                    # we won't call CellRange constructor which verifies that dep is an 'absolute' address
+                    # because 1) we don't use CellRange anymore (guess why), 2) It is verified and completed few lines before
                     if dep in cellmap:
+                        # case where dep (address) is well-formatted
                         # already dealt with this range
                         # add an edge from the range to the parent
-                        # print "Adding edge %s --> %s" % (rng.address(), c1.address())
-                        G.add_edge(cellmap[rng.address()], cellmap[c1.address()])
+                        # print "Adding edge %s --> %s" % (dep, c1.address())
+                        G.add_edge(cellmap[dep], cellmap[c1.address()])
                         continue
                     else:
 
-                        # TODO SHEEET
-                        my_cells, nrows, ncols = resolve_range(dep)
-                        my_cells = list(flatten(my_cells))
-
-                        values = [ self.cells[c].value for c in my_cells if c in self.cells]
-                        my_cells = [a for a in my_cells if a in self.cells]
-                        if len(values) != 0:
-                            # TODO what happens in this case ?
-                            my_range = Range((my_cells, nrows, ncols), values)
-
-                            self.ranges[dep] = my_range
-                            rng = Cell(dep, None, my_range, dep, True )
-                            self.cells[dep] = rng
-                            cellmap[dep] = rng
-
-                            # print "Adding edge %s --> %s" % (rng.address(), c1.address())
-                            G.add_edge(rng, cellmap[c1.address()])
-                            # cells in the range should point to the range as their parent
-                            target = self.cells[dep]
-                            cells = []
-                            for (child, value) in zip(my_cells, values):
-                                if child not in cellmap:
-                                    cells.append(Cell(child, None, value, self.cells[child].formula, False))
-                                
+                        # TODO SHEEET -> SBR: what you mean ?
+                        # TODO what happens in this case ? -> SBR: what you mean ?
```

Comment:
These comments were to remind to verify if in this case `Sheet` was provided, and if not, what happens.